electron/launcher/main-process/helpers/login.js

```diff
@@ -0,0 +1,203 @@
+const djangoUrl = config.DJANGO_SERVER.HOST +":" +config.DJANGO_SERVER.PORT;
+
+// REGULAR LOGIN WITH USERNAME AND PASSWORD
+const login = cred => {
+  return new Promise((resolve, reject) => {
+    let options = {
+      method: 'POST',
+      uri: djangoUrl +'/accounts/login/',
+      header: {'content-type': 'application/x-www-form-urlencoded'},
+      form: cred,
+      resolveWithFullResponse: true,
+      json: true
+    }
+    
+    let body;
+    let encryptedPassword;
+
+    aes256.encrypt(cred.password).then(encrypted => {
+      encryptedPassword = encrypted;
+    })
+    // SEND POST REQUEST TO LOGIN
+    .then(() => {
+      return request(options).then(response => {
+        body = response.body;
+        return response.headers['set-cookie'];
+      })
+      .catch(error => {
+        throw error.error;
+      });
+    })
+    // PARSE Set-Cookie HEADER INTO COOKIE OBJECTS
+    .then(cookieStrings => {
+      let promises = [];
+
+      for(cookieString of cookieStrings){
+        promises.push(new Promise((resolve, reject) => {
+          resolve(cookie.parse(cookieString));
+        }));
+      }
+
+      return Promise.all(promises).then(cookies => {
+        return cookies;
+      });
+    })
+    // SAVES USERNAME, PASSWORD, CSRF TOKEN, SESSION TOKEN IN OPERATING SYSTEM'S PASSWORD VAULT
+    .then(cookies => {
+      let promises = [];
+      let expiry = 0;
+
+      promises.push(keytar.setPassword('Dietstory', body.name, encryptedPassword));
+      for(cookieObject of cookies){
+        if(cookieObject.hasOwnProperty('csrftoken')){
+          csrfToken = cookieObject.csrftoken;   // GLOBAL VARIABLE
+          promises.push(keytar.setPassword('Dietstory_CSRF', body.name, cookieObject.csrftoken));
+        }
+        else if(cookieObject.hasOwnProperty('sessionid')){
+          sessionKey = cookieObject.sessionid;  // GLOBAL VARIABLE
+          promises.push(keytar.setPassword('Dietstory_Session', body.name, cookieObject.sessionid));
+          expiry = Date.parse(cookieObject.expires);
+        }
+      }
+      
+      return Promise.all(promises).then(() => {
+        return expiry;
+      })
+      .catch(error => {
+        throw error;
+      });
+    })
+    // SAVE USER INFO FOR FUTURE AUTO-LOGIN TO USE (ENCRYPTED)
+    .then(expiry => {
+      let encrypt = string => {
+        return new Promise((resolve, reject) => {
+          let cipher = crypto.createCipheriv(secret.algorithm, Buffer.from(secret.key, 'hex'), Buffer.from(secret.iv, 'hex'));
+          let encrypted = cipher.update(string, 'utf8', 'hex');
+          encrypted += cipher.final('hex');
+          resolve(encrypted);
+        });
+      }
+
+      sessionExpiry = expiry;   // GLOBAL VARIABLE
+      let userInfo = {...body, session_expiry: expiry};
+
+      return encrypt(JSON.stringify(userInfo)).then(encyrptedData => {
+        fs.writeFileSync(path.join(app.getPath('userData'), 'user_info'), encyrptedData, 'utf-8');
+        return userInfo;
```

Comment:
rename `encyrptedData` to `encryptedData`

Reply:
Also - why are we writing `user_data` to a file? Is this purely for caching purpose?

Reply:
Fixed typo. We are writing user_data to file because React will need user information if we want to skip the login request to Django. Django currently returns `username, email, is_active, is_superuser` upon successful login. It is encrypted to make it almost impossible to edit user_info and gain admin privileges.